cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/ToolCommand.java

```diff
@@ -228,28 +240,76 @@ private void handlePullCommand() {
             }
         }
 
-        Path balaCacheDirPath = ProjectUtils.createAndGetHomeReposPath()
-                .resolve(REPOSITORIES_DIR).resolve(CENTRAL_REPOSITORY_CACHE_NAME)
-                .resolve(ProjectConstants.BALA_DIR_NAME);
+        pullToolAndUpdateBalToolsToml(toolId, version);
+    }
 
-        if (isToolLocallyAvailable(toolId, version)) {
-            CommandUtil.printError(this.errStream, "tool is already pulled.", null, false);
+    private void handleUseCommand() {
+        if (argList.size() < 2) {
+            CommandUtil.printError(this.errStream, "no tool id given.", TOOL_USE_USAGE_TEXT, false);
+            CommandUtil.exitError(this.exitWhenFinish);
+            return;
+        }
+        if (argList.size() > 2) {
+            CommandUtil.printError(
+                    this.errStream, "too many arguments.", TOOL_USE_USAGE_TEXT, false);
             CommandUtil.exitError(this.exitWhenFinish);
             return;
         }
 
-        for (String supportedPlatform : SUPPORTED_PLATFORMS) {
-            try {
-                pullToolFromCentral(supportedPlatform, balaCacheDirPath);
-                insertToBalToolsTomlFile();
-            } catch (PackageAlreadyExistsException e) {
-                errStream.println(e.getMessage());
-                CommandUtil.exitError(this.exitWhenFinish);
-            } catch (CentralClientException e) {
-                errStream.println("unexpected error occurred while pulling tool:" + e.getMessage());
-                CommandUtil.exitError(this.exitWhenFinish);
-            }
+        String toolIdAndVersion = argList.get(1);
```

Comment:
Shall we introduce an inner class to hold the tool details? 
That way we can move this tool extraction logic to a util method and reuse it. WDYT?

Reply:
While it makes the code clean didn't  do so because it changes a lot of places and how we access the data in multiple places and that could possibly create new bugs. Perhaps will refactor before the next update along with the bal tool improvements